### ISSUE TITLE
"RAD_RcppRPCA" not available for .Call() for package "RAD"

### ISSUE THREAD

**Issue by ktpazder** [Netflix/Surus#20]
I cannot properly install this package and I think it may be Windows related. I have installed it using Surus, with the code provided in the README, and when I run AnomalyDetection.rpca I get the following error:

> Error in .Call("RAD_RcppRPCA", PACKAGE = "RAD", X, Lpenalty, Spenalty,  : 
>   "RAD_RcppRPCA" not available for .Call() for package "RAD"

The same procedure on a Mac works fine.

I also tried building Surus by cloning the project and using the "Clean & Build" option in RStudio. From that I receive the following error:

> ERROR: The build directory does not contain a DESCRIPTION
> file so cannot be built as a package.
> Build directory: ~/Bitbucket/tempsa/Surus

Am I missing something obvious?

**Comment by riff4** [comment]
Hey ! 

I'm currently facing the same issue and I'd be interested by your solution if you found one !

**Comment by ktpazder** [comment]
I did resolve this issue. The problem was that there exists another R package called RAD which stands for Rank Abundance Data. I had this package installed prior to installing the RPCA related RAD package.

It created dependency issues that couldn't be resolved until I removed the other RAD package entirely. But once removed, everything worked fine.

**Comment by triciascully** [comment]
I've removed the RAD package but I'm still getting the same error (I'm on a mac, don't think that matters but I thought I'd mention it since it was part of the convo before!). I went to my lib/R/library directory to check that RAD was uninstalled just to be sure, and it's gone, but the error keeps popping up. Any ideas on what I'm doing wrong would be much appreciated!

**Comment by ktpazder** [comment]
You're right in that the operating system had nothing to do with it. But all that I needed to do was remove the old package and then install this Surus/RAD package using devtools (code below). I don't have any guidance beyond this.

library(devtools)
install_github(repo = "Surus", username = "Netflix", subdir = "resources/R/RAD")

**Comment by triciascully** [comment]
Thank you! That helped with my initial error, but then I started getting:

Error in ggplot_AnomalyDetection.ma(AnomalyDetection.ma(ts)) : 
  could not find function "ggplot_AnomalyDetection.ma"

I terminated R, reran everything (the code you provided), still saw the error, then manually ran the raw code provided for the function (which fixed the error) but then I got this error:

Error in multiplot(p1, p2, p3, p4, cols = 2) : 
  could not find function "multiplot"

I now have the raw R code that creates all the functions as well as the code you provided loading the necessary packages before the test1 code, which seems to work, but it also seems like overkill. Not sure if there's a bug or if dependencies have been updated that might be causing the error - will update my comments if I figure it out, but it's working for the time being!
